distributions/test/src/test/java/com/connexta/ion/replication/distributions/test/ReplicationITest.java

```diff
@@ -0,0 +1,144 @@
+      sourceUrl = "https://ditto-1.phx.connexta.com:8993/services";
+    }
+    if (destinationUrl == null) {
+      destinationUrl = "https://ditto-2.phx.connexta.com:8993/services";
+    }
+  }
+
+  @After
+  public void cleanUp() {
+    idsCreated.forEach(this::deleteFromNodes);
+    clearAllSolrCores();
+  }
+
+  @Test
+  public void createXmlFile() throws Exception {
+    LOGGER.info("Solr URL: {}", solrUrl);
+    String testId = UUID.randomUUID().toString();
+
+    // create product
+    String id =
+        createProductOnSourceNode(
+            APPLICATION_XML, "data/reuters-xml-sample", ImmutableMap.of("testId", testId));
+    LOGGER.info("Returned ID: {}", id);
+    idsCreated.add(id);
+
+    // create nodes and jobs
+    postToSolrCore(
+        SITE, "sites.json", ImmutableMap.of("source", sourceUrl, "destination", destinationUrl));
+    postToSolrCore(CONFIG, "jobs.json", ImmutableMap.of("testId", testId));
+
+    // wait for product to be replicated
+    await()
+        .with()
+        .pollInterval(1, TimeUnit.SECONDS)
+        .atMost(40, TimeUnit.SECONDS)
+        .until(queryByIdSucceeds(id));
+
+    // confirm metacard has resource
+    XmlPath xmlPath =
+        given()
+            .relaxedHTTPSValidation()
+            .when()
+            .get(destinationUrl + CATALOG_ENDPOINT + "/" + id)
+            .then()
+            .extract()
+            .body()
+            .xmlPath();
+
+    // TODO: should I verify the replication origins attribute is added?
+    // this will make it more difficult to make the source and destination configurable
+    assertTrue(
+        xmlPath
+            .getList("metacard.string.find { it.@name == 'metacard-tags' }.value", String.class)
+            .contains("replicated"));
+    assertTrue(
+        xmlPath
+            .getString("metacard.string.find { it.@name == 'title' }.value[0]")
+            .contains(testId));
+  }
+
+  private void postToSolrCore(
+      String core, String jsonFilePath, ImmutableMap<String, String> varReplacementMap) {
+    given()
+        .header(CONTENT_TYPE, APPLICATION_JSON)
+        .body(getFileAsText(jsonFilePath, varReplacementMap))
+        .when()
+        .post(solrCoreUpdateUrl(core))
+        .then()
+        .statusCode(200);
+  }
+
+  private void deleteFromNodes(String id) {
+    given().relaxedHTTPSValidation().when().delete(sourceUrl + CATALOG_ENDPOINT + "/" + id);
+    given().relaxedHTTPSValidation().when().delete(destinationUrl + CATALOG_ENDPOINT + "/" + id);
+  }
+
+  private String createProductOnSourceNode(
+      String contentType, String resourcePath, ImmutableMap<String, String> varReplacementMap) {
+    return given()
+        .relaxedHTTPSValidation()
+        .header(CONTENT_TYPE, contentType)
+        .body(getFileAsText(resourcePath, varReplacementMap))
+        .expect()
+        .log()
+        .headers()
+        .statusCode(201)
+        .when()
+        .post(sourceUrl + CATALOG_ENDPOINT)
+        .getHeader("id");
```

Comment:
❓ Do we have a guarantee that the product is ready to be queried on the source after this call returns 201?

Reply:
I think so since we are querying by ID. DDF will cache uncommitted entries in a near real time cache which will be queried only if it is a query by id.

Reply:
That means we are relying on that specific implementation detail. What is DDF is using SolrCloud? 
Should we be more cautious and safe by making sure it can be queried before we move on?